Add deprecation utilities

hesseltuinhof opened the pull request:
# Description

Add deprecation decorators to ART. This will make it easier to deprecate functions, methods and keyword arguments.

Related to #423 

## Type of change

- [x] New feature (non-breaking)
- [x] This change requires a documentation update

# Testing

Several test cases for both decorators have been included.

**Test Configuration**:
- Fedora 31
- Python 3.6.10
- ART 1.3.0 development

# Checklist

- [x] My code follows the style guidelines of this project
- [x] I have performed a self-review of my own code
- [x] I have commented my code
- [x] I have made corresponding changes to the documentation
- [x] My changes generate no new warnings
- [x] I have added tests that prove my fix is effective or that my feature works
- [x] New and existing unit tests pass locally with my changes

beat-buesser commented on art/utils.py:
```diff
@@ -20,15 +20,102 @@
 logger = logging.getLogger(__name__)
 
 
+def deprecated(end_version, *, reason="", replaced_by=""):
```
```suggestion
# ------------------------------------------------------------------------------------------------- DEPRECATION
def deprecated(end_version, *, reason="", replaced_by=""):
```

hesseltuinhof commented:
Thanks for the review @beat-buesser. I have incorporated your suggested change. Furthermore I noticed that I forgot to add the `tests/utils` to the test runner. I have fixed this also.

beat-buesser commented:
@hesseltuinhof Thank you, great catch!
